packages/ai/core/util/detect-audio-mime-type.ts

```diff
@@ -0,0 +1,56 @@
+import { DataContent } from '../prompt';
+import { convertDataContentToUint8Array } from '../prompt/data-content';
+
+const mimeTypeSignatures = [
+  {
+    mimeType: 'audio/mpeg' as const,
+    bytesPrefix: [0xff, 0xfb],
+    base64Prefix: 'SUQzBA',
+  },
+  {
+    mimeType: 'audio/wav' as const,
+    bytesPrefix: [0x52, 0x49, 0x46, 0x46],
+    base64Prefix: 'UklGR',
+  },
+  {
+    mimeType: 'audio/ogg' as const,
+    bytesPrefix: [0x4f, 0x67, 0x67, 0x53],
+    base64Prefix: 'T2dnUw',
+  },
+  {
+    mimeType: 'audio/flac' as const,
+    bytesPrefix: [0x66, 0x4c, 0x61, 0x43],
+    base64Prefix: 'ZkxhQw',
+  },
+  {
+    mimeType: 'audio/aac' as const,
+    bytesPrefix: [0x40, 0x15, 0x00, 0x00],
+    base64Prefix: 'AAC',
+  },
+  {
+    mimeType: 'audio/mp4' as const,
+    bytesPrefix: [0x66, 0x74, 0x79, 0x70],
+    base64Prefix: 'AAAA',
+  },
+] as const;
+
+export function detectAudioMimeType(
+  audio: DataContent,
+): (typeof mimeTypeSignatures)[number]['mimeType'] | undefined {
+  const uint8ArrayAudio = convertDataContentToUint8Array(audio);
+
+  for (const signature of mimeTypeSignatures) {
+    if (
+      typeof audio === 'string'
+        ? audio.startsWith(signature.base64Prefix)
+        : uint8ArrayAudio.length >= signature.bytesPrefix.length &&
+          signature.bytesPrefix.every(
+            (byte, index) => uint8ArrayAudio[index] === byte,
+          )
+    ) {
+      return signature.mimeType;
+    }
+  }
+
+  return undefined;
+}
```

Comment:
extract a generalized, generic `detectMimeType` that receives `data` and `signatures` (the array, generic) and use it from both the audio and image version. Tests could also be shifted to the generic version, test for audio/image can then be removed.